### PR TITLE
[2099] Update UCAS contacts

### PR DESCRIPTION
**Pull request by davidgisbey**
### Context

Providers needs to be able to update their UCAS contacts

### Changes proposed in this pull request

- Add the five UCAS contacts to provider #update

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by davidgisbey** [comment]
[Edit UCAS apply contacts](https://trello.com/c/Xec5lY1e/2099-edit-ucas-apply-contacts)

File: app/controllers/api/v2/providers_controller.rb [hunk 1]
```diff
@@ -124,6 +125,16 @@ def update_enrichment
         enrichment.save
       end
 
+      def update_ucas_contacts
+        return if ucas_contact_params.blank?
+
+        ucas_contact_params.keys.each do |type|
+          contact = @provider.contacts.find_or_create_by(type: type.gsub(/_contact$/, ''))
```
Comment (misaka) [inline app/controllers/api/v2/providers_controller.rb]:
```suggestion
          contact = @provider.contacts.find_or_initialize_by(type: type.gsub(/_contact$/, ''))
```

Comment (davidgisbey) [reply app/controllers/api/v2/providers_controller.rb]: Done

File: app/controllers/api/v2/providers_controller.rb [hunk 2]
```diff
@@ -124,6 +125,16 @@ def update_enrichment
         enrichment.save
       end
 
+      def update_ucas_contacts
+        return if ucas_contact_params.blank?
+
+        ucas_contact_params.keys.each do |type|
+          contact = @provider.contacts.find_or_initialize_by(type: type.gsub(/_contact$/, ''))
```
Comment (timabell) [inline app/controllers/api/v2/providers_controller.rb]: I was wondering about user input on `gsub` so I asked the internet https://security.stackexchange.com/q/216816/158425

**Pull request merged**